sktime/transformations/series/fourier.py

```diff
@@ -23,23 +23,27 @@ class FourierFeatures(BaseTransformer):
         - sin_sp_k = :math:`sin(\frac{2 \pi k t}{sp})`
         - cos_sp_k = :math:`cos(\frac{2 \pi k t}{sp})`
 
-    Where :math:`t` is the number of time steps elapsed from the beginning of the time
-    series.
+    Where :math:`t` is the elapsed time since the beginning of the seasonal period and
+    :math:`sp` the total time of the seasonal period.
 
-    The output of the transform is a pandas DataFrame that includes the fourier terms as
+    The transformed output is a pandas DataFrame that includes the fourier terms as
     additional columns with the naming convention stated above (sin_sp_k and cos_sp_k).
-    For instance for sp_list = [12, 3] and fourier_terms_list = [2, 1] the transformed
+    For instance for sp_list = [12, "Y"] and fourier_terms_list = [2, 1] the transformed
     series will have the additional columns:
-    "cos_12_1", "sin_12_1", "cos_12_2", "sin_12_2", "cos_3_1", "sin_3_1"
+    "cos_12_1", "sin_12_1", "cos_12_2", "sin_12_2", "cos_Y_1", "sin_Y_1"
```

Comment:
something isn't quite right here. Is this the paired product or the cross-product?

If we pair 12, Y and 2,1, then we get (12, 2) and (Y, 1), so should we not get `"cos_12_2", "sin_12_2", "cos_Y_1", "sin_Y_1"` only?

Whereas if we cross-product them, we get (12, 2), (12, 1), (Y, 2), (Y, 1), so should we not get `"cos_12_2", "sin_12_2", "cos_12_1", "sin_12_1", "cos_Y_2", "sin_Y_2", "cos_Y_1", "sin_Y_1"`, in this order?

Neither plausible options seems to give the claimed columns.

Reply:
thanks for the explainer, clear now.